tests/integ/test_dataframe.py

```diff
@@ -416,6 +425,61 @@ def process(self, n: int) -> int:
     Utils.check_answer(df.select(df.a, table_func(df.b, lit(" "))), expected_result)
 
 
+def test_generator_table_function(session):
+    # works with rowcount
+    expected_result = [Row(-108, 3), Row(-107, 3), Row(0, 3)]
+    df = (
+        session.generator(seq1(1), uniform(1, 10, 2), rowcount=150)
+        .order_by(seq1(1))
+        .limit(3, offset=20)
+    )
+    Utils.check_answer(df, expected_result)
+
+    # works with timelimit
+    expected_result = [Row(-128, 3), Row(-128, 3), Row(-128, 3)]
+    df = (
+        session.generator(seq1(1), uniform(1, 10, 2), timelimit=1)
+        .order_by(seq1(1))
+        .limit(3)
+    )
+    Utils.check_answer(df, expected_result)
+
+    # works with combination of both
+    expected_result = [Row(-108, 3), Row(-107, 3), Row(0, 3)]
+    df = (
+        session.generator(seq1(1), uniform(1, 10, 2), timelimit=1, rowcount=150)
+        .order_by(seq1(1))
+        .limit(3, offset=20)
+    )
+    Utils.check_answer(df, expected_result)
+
+    # works without both
+    df = session.generator(seq1(1), uniform(1, 10, 2))
+    Utils.check_answer(df, [])
+
+    # aliasing works
+    df = (
+        session.generator(
+            seq1(1).as_("pixel"), uniform(1, 10, 2).as_("unicorn"), rowcount=150
+        )
+        .order_by("pixel")
+        .limit(3, offset=20)
+    )
+    expected_result = [
+        Row(pixel=-108, unicorn=3),
+        Row(pixel=-107, unicorn=3),
+        Row(pixel=0, unicorn=3),
+    ]
+    Utils.check_answer(df, expected_result)
+
```

Comment:
can we also have a test case `session.generator(count(seq1()), ...)` (just something with aggregation function)?